kubernetes-repositories.md

```diff
@@ -0,0 +1,106 @@
+## Kubernetes Repositories
+
+
+This document attempts to describe a process for creating and associating github repositories with the Kubernetes project.
+
+The document presents a tiered system of repositories with increasingly strict requirements in an attempt to provide the right level of oversight and flexibility for a variety of different projects.
+
+
+### Associated Repositories
+
+Associated repositories conform to the Kubernetes community standards for a repository, but otherwise have no restrictions. Associated repositories exist solely for the purpose of making it easier for the Kubernetes community to work together. There is no implication of support or endorsement of any kind by the Kubernetes project, the goals are purely logistical. 
+
+#### Goals
+
+To facilitate contributions and collaboration from the broader Kubernetes community.  Contributions to random projects with random CLAs (or DCOs) can be logistically difficult, so associated repositories should be easier.
+
+
+#### Rules
+
+   * Must adopt the Kubernetes Code of Conduct statement in their repo.
+   * All code projects use the Apache License version 2.0. Documentation repositories should use the Creative Commons License version 4.0.
+   * Must adopt the CNCF CLA bot automation for pull requests.
+
+
+### SIG repositories
+
+SIG repositories serve as temporary homes for SIG-sponsored experimental projects or prototypes of new core functionality, or as permanent homes for SIG-specific tools.
+
+#### Goals
+
+To provide a place for SIGs to collaborate on projects endorsed by and actively worked on by members of the SIG. SIGs should be able to approve and create new repositories for SIG-sponsored projects without requiring higher level approval from a central body (e.g. steering committee or sig-architecture)
+
+#### Rules
+
+   * For now all repos will live in github.com/kubernetes-sigs/<sig-name>-<project-name>, this may change in the future as we see how things work.
+   * Must adopt the Kubernetes Code of Conduct
+   * All code projects use the Apache License version 2.0. Documentation repositories should use the Creative Commons License version 4.0.
+   * Must adopt the CNCF CLA bot, merge bot and Kubernetes PR commands/bots.
```

Comment:
Why must all repos adopt the merge automation and PR commands/bots (I assume associated with merge automation)?

I ask due to contributor experience. k8s merge automation is not common. It solves a real problem for the monorepo but it's not common among k8s repos or repos outside of k8s and many folks who are involved in existing k8s repos, many of whom don't use the merge automation, don't understand how it works. If you use it regularly it makes sense but that's a subset of people.

From my perspective, I keep getting asked how it works. Sometimes folks I have explained it to come back and ask a couple months later because it's been that long since they dealt with it.

If everyone is going to have to use it the UX of the solution needs to be improved so the long tail of folks can navigate it.

Reply:
We want consistent experience for all projects that are associated with Kubernetes, for example two-factor auth, merge via bot/test rather than human, etc.

Given that you can always build an associated repository which has lighter responsibility.